app/services/transcript_service.rb

```diff
@@ -23,6 +23,11 @@ def reset
     end
   end
 
+  # always return a record
+  def self.find_by_uid(uid)
+    Transcript.published.find_by(uid: uid) || Transcript.new
```

Comment:
Since you are finding by a specific uid, if it doesn't exist shouldn't the new one have the uid set to the one given to this method?

Reply:
@ReganRyanNZ , thanks, In this case, no, because, the reason for this change is , not to return nil and return a blank object. 

Then the callee (the frontend (backbone)) in this case will handle the blank object

Reply:
👍

Reply:
@sameera207 I wonder if it would be better to have that logic in the caller rather than in the fine_by_uid?
Or rename the method to indicate that it may return an empty nil object? - but that would be a long name?

Reply:
@craighawkes , good point, but in the case the caller is a frontend backbone app. Which makes the changes bit harder. So, this will be a safer way with less code change. 

RE: method name, Yes, I remember sandi matz has a video and she gives a name like `guranteed_object`, but yes, I should come up with more meaningful name next time :). 

[There are only two hard things in Computer Science: cache invalidation and naming things.](https://martinfowler.com/bliki/TwoHardThings.html)